flocker/control/test/test_protocol.py

```diff
@@ -137,7 +141,39 @@ def build_control_amp_service(test):
                              TCP4ServerEndpoint(MemoryReactor(), 1234))
 
 
-class ControlAMPTests(SynchronousTestCase):
+class ControlTestCase(SynchronousTestCase):
+    """
+    Base TestCase for control tests that supplies a utility
+    method to patch the callRemote method of the AMP protocol instance,
+    discarding Eliot contexts whose context level may be unknown.
+    """
+
+    def patch_call_remote(self, capture_list, protocol):
+        """
+        Patch the callRemote method for this test case's protocol.
+
+        :param capture_list: A `list` to which results will be added.
+
+        :param protocol: Either `None` to default to self.protocol, or
+            a ``ControlAMP`` instance.
+        """
+
+        def capture_call_remote(capture, *args, **kwargs):
+            # Ditch the eliot context whose context level is difficult to
+            # predict.
+            kwargs.pop('eliot_context')
+            capture.append((args, kwargs))
+            return succeed(None)
+
+        self.patch(
+            protocol,
+            "callRemote",
+            lambda *args, **kwargs: capture_call_remote(
+                capture_list, *args, **kwargs)
+        )
```

Comment:
This ... is just refactoring of existing code, I guess.  This (the pre-existing code or the refactored code) is really not how we want to be writing tests though.  This is a tedious, error prone way to test.  It's another semi-re-implementation of the mock testing pattern.  The exact arguments passed to `callRemote` aren't the important thing.  The behavior achieved is.  This level of testing only exercises the former, though.

This is a pre-existing issue so it doesn't block this branch, though.